barretenberg/cpp/src/barretenberg/sumcheck/sumcheck.hpp

```diff
@@ -173,12 +174,21 @@ template <typename Flavor, const size_t virtual_log_n = CONST_PROOF_SIZE_LOG_N>
     */
     PartiallyEvaluatedMultivariates partially_evaluated_polynomials;
     // prover instantiates sumcheck with circuit size and a prover transcript
-    SumcheckProver(size_t multivariate_n, const std::shared_ptr<Transcript>& transcript)
+    SumcheckProver(size_t multivariate_n,
+                   const std::shared_ptr<Transcript>& transcript,
+                   RelationSeparator& relation_separator)
         : multivariate_n(multivariate_n)
         , multivariate_d(numeric::get_msb(multivariate_n))
         , transcript(transcript)
-        , round(multivariate_n){};
+        , round(multivariate_n)
+        , alphas(std::move(relation_separator)){};
```

Comment:
for this to be useful you would need to use std::move in the invocation of the constructor (which I haven't seen so far) and change the parameter type to RelationSeparator&&. I'm not sure how others feel about this but my preference is generally to avoid moving things unless its clearly relevant for performance so I think a reference/copy is just fine here. up to you

Reply:
Also I'm not sure std::move works with an lvalue reference (one `&`) IIRC you need an rvalue reference (two `&&`) or just a non-trivially copyable object.

Reply:
btw, shared pointers should always be copied and moved e.g. taken as param as `std::shared_ptr<Transcript> transcript` and then `transcript(std::move(transcript))`. This avoids one copy, and one global lock unlock.